src/hotspot/share/gc/shenandoah/heuristics/shenandoahAdaptiveHeuristics.cpp

```diff
@@ -110,6 +204,30 @@ bool ShenandoahAdaptiveHeuristics::should_start_gc() const {
   size_t soft_tail = max_capacity - capacity;
   available = (available > soft_tail) ? (available - soft_tail) : 0;
 
+  size_t bytes_allocated_since_gc_start = heap->bytes_allocated_since_gc_start();
+
+  // Track allocation rate even if we decide to start a cycle for other reasons.
+  ShenandoahAdaptiveHeuristics *heuristic = const_cast<ShenandoahAdaptiveHeuristics *>(this);
+  heuristic->_allocation_rate.sample(bytes_allocated_since_gc_start);
+  heuristic->_last_trigger = OTHER;
```

Comment:
Shouldn't `_last_trigger` updates be near `return true;` in this method? I.e. we should not probably overwrite `_last_trigger` if nothing was actually triggered? This probably affects the super-call to `ShenandoahHeuristics::should_start_gc()` as well?

Reply:
Would it make more sense if I `OTHER` were `NONE`? The intention here is to only adjust the allocation rate or spike detection parameters if they triggered the cycle. Here we reset the `_last_trigger` in one place without having to touch all the actual decision points. As it stands, once a gc cycle is started, the control thread won't call back into this method until the cycle is complete.

Reply:
(realized I have not replied here) `OTHER` is fine. `NONE` implies no trigger at all, which is a lie when some trigger actually acts :)